app/models/volunteer.rb

```diff
@@ -1,2 +1,5 @@
 class Volunteer < ApplicationRecord
+  validates :email, :password, :salt, presence: true # all these fields are required
+  validates_format_of :email, :with => /\A[^@\s]+@([^@\s]+\.)+[^@\s]+\z/ # email must have correct format
```

Comment:
The parens in the regex here are no necessary here as there isn't a group needed for substitution